centreon/tests/e2e/features/Dashboards/18-web-page-widget-configuration/index.ts

```diff
@@ -0,0 +1,185 @@
+import { Given, When, Then } from '@badeball/cypress-cucumber-preprocessor';
+
+import dashboardAdministratorUser from '../../../fixtures/users/user-dashboard-administrator.json';
+import dashboards from '../../../fixtures/dashboards/creation/dashboards.json';
+import webPageWidget from '../../../fixtures/dashboards/creation/widgets/dashboardWithWebPageWidget.json';
+import webPageWidgets from '../../../fixtures/dashboards/creation/widgets/dashboardWithTwoWebPageWidget.json';
+
+const validUrl = 'https://docs.centreon.com/fr/';
+const invalidUrl = 'http://docss.Centreon.com/fr/';
+const iframeContent = 'Bienvenue dans la Documentation Centreon !';
+
+before(() => {
+  cy.intercept({
+    method: 'GET',
+    url: '/centreon/api/internal.php?object=centreon_topology&action=navigationList'
+  }).as('getNavigationList');
+  cy.intercept({
+    method: 'GET',
+    url: /\/centreon\/api\/latest\/monitoring\/resources.*$/
+  }).as('resourceRequest');
+  cy.startContainers();
+  cy.enableDashboardFeature();
+  cy.executeCommandsViaClapi(
+    'resources/clapi/config-ACL/dashboard-metrics-graph.json'
+  );
+  cy.applyAcl();
+});
+
+beforeEach(() => {
+  cy.intercept({
+    method: 'GET',
+    url: '/centreon/api/internal.php?object=centreon_topology&action=navigationList'
+  }).as('getNavigationList');
+  cy.intercept({
+    method: 'GET',
+    url: '/centreon/api/latest/configuration/dashboards**'
+  }).as('listAllDashboards');
+  cy.intercept({
+    method: 'PATCH',
+    url: `/centreon/api/latest/configuration/dashboards/*`
+  }).as('updateDashboard');
+  cy.intercept({
+    method: 'GET',
+    url: `/centreon/api/latest/configuration/dashboards/*`
+  }).as('getDashboard');
+  cy.intercept({
+    method: 'GET',
+    url: /\/api\/latest\/monitoring\/dashboard\/metrics\/performances\/data\?.*$/
+  }).as('performanceData');
+  cy.intercept({
+    method: 'GET',
+    url: /\/centreon\/api\/latest\/monitoring\/resources.*$/
```

Comment:
I don't know if it's done like this everywhere, but wouldn't it be better (cleaner and more maintainable) to extract the urls into dedicated files or constants? Because it's easier to use the same url, or start of url, several times than to type it everywhere.

Reply:
As @MedWassCh is on holiday, I can't answer your question at the moment, but I will inform him when he returns

Reply:
I understand the idea of extracting URLs into files or constants, but in this context, it’s not necessarily the best practice. Adding abstraction for URLs unnecessarily complicates Cypress tests, which are designed to be simple and readable. Keeping the URLs directly in the intercepts makes the tests easier to understand and maintain, especially when the endpoints are specific or dynamic. Centralizing URLs can introduce reuse errors, complicate test management, and reduce flexibility for quickly adjusting tests without impacting other scenarios.